pkg/azuredisk/nodeserver_v2.go

```diff
@@ -80,11 +64,19 @@ func (d *DriverV2) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolume
 	}
 	defer d.volumeLocks.Release(diskURI)
 
-	lunStr, ok := req.PublishContext[consts.LUN]
+	publishContext, err := d.crdProvisioner.WaitForAttach(ctx, diskURI, d.NodeID, consts.CSITimeout/2)
```

Comment:
We should rely on the deadline set in the context for timeouts and if we need to make the timeout different (i.e. possibly shorter), create a new context from the `WithTimeout()`, e.g.
```go
waitCtx, waitCancel := context.WithTimeout(ctx, consts.CSITimeout/2)
defer cancel()

publishContext, err := d.crdProvisioner.WaitForAttach(waitCtx, diskURI, d.NodeID)
```

Still using a 1 minute timeout here is probably not the right thing to do given we've seen P99 attach latencies in the 2 minute range.